src/main/java/org/wso2/carbon/inbound/custom/poll/MQConnectionBuilder.java

```diff
@@ -0,0 +1,243 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.inbound.custom.poll;
+
+import com.ibm.mq.MQEnvironment;
+import com.ibm.mq.MQException;
+import com.ibm.mq.MQQueueManager;
+import com.ibm.mq.MQSimpleConnectionManager;
+import com.ibm.mq.constants.CMQC;
+import com.ibm.mq.constants.CMQXC;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.FileInputStream;
+import java.security.KeyStore;
+import java.util.Collection;
+import java.util.Properties;
+import java.util.Stack;
+import java.util.Vector;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Start connection with IBM MQ queue manager
+ */
+public class MQConnectionBuilder {
```

Comment:
Please make this class a utility class MQConnectionUtils and remove all the class level variable. You need to address all stuff discussed in the IBM-MQ connector PR. [1]
[1] https://github.com/wso2-incubator/esb-connector-ibmmq/pull/1